api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMedicationRequestDaoImpl.java

```diff
@@ -14,16 +14,59 @@
 import lombok.AccessLevel;
 import lombok.Setter;
 import org.hibernate.Criteria;
+import org.hibernate.criterion.Restrictions;
 import org.openmrs.DrugOrder;
+import org.openmrs.Order;
 import org.openmrs.module.fhir2.FhirConstants;
 import org.openmrs.module.fhir2.api.dao.FhirMedicationRequestDao;
 import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
 import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Component
 @Setter(AccessLevel.PACKAGE)
 public class FhirMedicationRequestDaoImpl extends BaseFhirDao<DrugOrder> implements FhirMedicationRequestDao {
 	
+	@Override
```

Comment:
For the two "get" and the "getSearcResults" methods I went back and forth on replacing the BaseFhirDAO entirely with a new query versus delegating to them and then stripping out any "discontinue" orders.  I went with the second approach (delegating and stripping) because it seemed less intrusive to the original code, and, also, the session factory currently has private access and so wasn't available in this extension class.

Reply:
Just an observation, but while the `sessionFactory` itself is a private field, there is a public getter, so you should be able to use `getSessionFactory()` factory here to get a hold of it (of course, this depends on your IDE having the appropriate extension for Lombok or else I think it's kind of hidden).

Reply:
duh, good point @ibacher .... (still think I will stick the current approach I've gone with, though)